File: 백준/1325/solution.py
import sys, collections
input = sys.stdin.readline
N, M = map(int, input().split())

dic = collections.defaultdict(list)
for _ in range(M):
    a, b = map(int, input().split())
    dic[b].append(a)

def bfs(start):
    visited = [0] * (N+1)
    q = collections.deque()
    q.append(start)
    visited[start] = 1
    cnt = 0
    while q:
        cur = q.popleft()
        for nxt in dic[cur]:
            if visited[nxt] == 0:
                cnt += 1
                visited[nxt] = 1
                q.append(nxt)
    return cnt

counts = [0] * (N+1)
for i in range(1, N+1):
    counts[i] = bfs(i)
maxCount = max(counts)
for i in range(1, N+1):
    if counts[i] == maxCount:
        print(i, end=' ')

File: 백준/1325/solution.swift
import Foundation

// Input
let NM = readLine()!.split(separator: " ").compactMap{Int($0)}
let N = NM[0]
let M = NM[1]
var graph = [[Int]](repeating: [], count: N+1)
for _ in 0..<M {
    let AB = readLine()!.split(separator: " ").compactMap{ Int($0) }
    let A = AB[0]
    let B = AB[1]
    graph[B].append(A)
}
print(graph)

// BFS / DFS

// Solve